### ISSUE TITLE
REUSEPORT option not working 

### ISSUE THREAD

**Issue by thechile** [PowerDNS/pdns#5257]
I am using the latest pdns 4.x with `reuseport=yes` configured but it doesn't appear to work when using the default `pdns` user. If i change `setuid=root` and then `systemctl restart pdns` it will work as expected.

```
#packages
pdns-4.0.3-1pdns.el7.x86_64
pdns-backend-mysql-4.0.3-1pdns.el7.x86_64
pdns-backend-geoip-4.0.3-1pdns.el7.x86_64
pdns-recursor-4.0.4-1pdns.el7.x86_64

#os
lsb_release -a
LSB Version:    :core-4.1-amd64:core-4.1-noarch:cxx-4.1-amd64:cxx-4.1-noarch:desktop-4.1-amd64:desktop-4.1-noarch:languages-4.1-amd64:languages-4.1-noarch:printing-4.1-amd64:printing-4.1-noarch
Distributor ID: CentOS
Description:    CentOS Linux release 7.3.1611 (Core)
Release:        7.3.1611
Codename:       Core

#kernel
Linux 3.10.0-514.16.1.el7.x86_64 #1 SMP Wed Apr 12 15:04:24 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

#selinux
sestatus
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   permissive             <=====
Mode from config file:          permissive
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Max kernel policy version:      28

```

Below is log output when using the default `setuid=pdns` in pdns.conf:

```
Apr 20 04:36:50 Done launching threads, ready to distribute questions
Apr 20 04:36:50 Starting new listen thread on the same IPs/ports using SO_REUSEPORT
Apr 20 04:36:50 Done launching threads, ready to distribute questions
Apr 20 04:36:50 Starting new listen thread on the same IPs/ports using SO_REUSEPORT
Apr 20 04:36:50 Done launching threads, ready to distribute questions
Apr 20 04:36:50 Unable to bind UDP socket to 'x.x.x.x:53': Permission denied
Apr 20 04:36:50 Unable to bind UDP socket to 'x.x.x.x:53': Permission denied
Apr 20 04:36:50 Unable to reuse port, falling back to original bind
Apr 20 04:36:50 Unable to reuse port, falling back to original bind
Apr 20 04:36:50 Done launching threads, ready to distribute questions
Apr 20 04:36:50 Starting new listen thread on the same IPs/ports using SO_REUSEPORT
Apr 20 04:36:50 Unable to bind UDP socket to 'x.x.x.x:53': Permission denied
Apr 20 04:36:50 Unable to reuse port, falling back to original bind
Apr 20 04:36:50 Done launching threads, ready to distribute questions
Apr 20 04:36:50 Starting new listen thread on the same IPs/ports using SO_REUSEPORT
Apr 20 04:36:50 Unable to bind UDP socket to 'x.x.x.x:53': Permission denied
Apr 20 04:36:50 Unable to reuse port, falling back to original bind
Apr 20 04:36:50 Done launching threads, ready to distribute questions
Apr 20 04:36:50 Starting new listen thread on the same IPs/ports using SO_REUSEPORT
Apr 20 04:36:50 Unable to bind UDP socket to 'x.x.x.x:53': Permission denied
Apr 20 04:36:50 Unable to reuse port, falling back to original bind
```

And here is strace network output:

```
Apr 20 05:33:24 Done launching threads, ready to distribute questions
Apr 20 05:33:24 Starting new listen thread on the same IPs/ports using SO_REUSEPORT
[pid  4614] socket(PF_INET, SOCK_DGRAM, IPPROTO_IP) = 123
[pid  4614] setsockopt(123, SOL_SOCKET, SO_TIMESTAMP, [1], 4) = 0
[pid  4614] setsockopt(123, SOL_SOCKET, SO_REUSEPORT, [1], 4) = 0
[pid  4614] bind(123, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("x.x.x.x")}, 16) = -1 EACCES (Permission denied)
Apr 20 05:33:24 Unable to bind UDP socket to 'x.x.x.x:53': Permission denied
Apr 20 05:33:24 Unable to reuse port, falling back to original bind
Apr 20 05:33:24 Done launching threads, ready to distribute questions
Apr 20 05:33:24 Done launching threads, ready to distribute questions
Apr 20 05:33:24 Starting new listen thread on the same IPs/ports using SO_REUSEPORT
[pid  4616] socket(PF_INET, SOCK_DGRAM, IPPROTO_IP) = 123
[pid  4616] setsockopt(123, SOL_SOCKET, SO_TIMESTAMP, [1], 4) = 0
[pid  4616] setsockopt(123, SOL_SOCKET, SO_REUSEPORT, [1], 4) = 0
[pid  4616] bind(123, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("x.x.x.x")}, 16Apr 20 05:33:24 Done launching threads, ready to distribute questions
) = -1 EACCES (Permission denied)
Apr 20 05:33:24 Starting new listen thread on the same IPs/ports using SO_REUSEPORT
[pid  4619] socket(PF_INET, SOCK_DGRAM, IPPROTO_IPApr 20 05:33:24 Unable to bind UDP socket to 'x.x.x.x:53': Permission denied
) = 123
Apr 20 05:33:24 Unable to reuse port, falling back to original bind
[pid  4619] setsockopt(123, SOL_SOCKET, SO_TIMESTAMP, [1], 4) = 0
[pid  4619] setsockopt(123, SOL_SOCKET, SO_REUSEPORT, [1], 4) = 0
[pid  4619] bind(123, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("x.x.x.x")}, 16) = -1 EACCES (Permission denied)
Apr 20 05:33:24 Unable to bind UDP socket to 'x.x.x.x:53': Permission denied
Apr 20 05:33:24 Unable to reuse port, falling back to original bind
Apr 20 05:33:24 Done launching threads, ready to distribute questions
Apr 20 05:33:24 Starting new listen thread on the same IPs/ports using SO_REUSEPORT
[pid  4615] socket(PF_INET, SOCK_DGRAM, IPPROTO_IP) = 123
Apr 20 05:33:24 Done launching threads, ready to distribute questions
[pid  4615] setsockopt(123, SOL_SOCKET, SO_TIMESTAMP, [1], 4) = 0
[pid  4615] setsockopt(123, SOL_SOCKET, SO_REUSEPORT, [1], 4Apr 20 05:33:24 Starting new listen thread on the same IPs/ports using SO_REUSEPORT
) = 0
[pid  4620] socket(PF_INET, SOCK_DGRAM, IPPROTO_IP <unfinished ...>
[pid  4615] bind(123, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("x.x.x.x")}, 16) = -1 EACCES (Permission denied)
[pid  4620] <... socket resumed> )      = 124
[pid  4620] setsockopt(124, SOL_SOCKET, SO_TIMESTAMP, [1], 4) = 0
[pid  4620] setsockopt(124, SOL_SOCKET, SO_REUSEPORT, [1], 4) = 0
Apr 20 05:33:24 Unable to bind UDP socket to 'x.x.x.x:53': Permission denied
[pid  4620] bind(124, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("x.x.x.x")}, 16) = -1 EACCES (Permission denied)
Apr 20 05:33:24 Unable to reuse port, falling back to original bind
Apr 20 05:33:24 Unable to bind UDP socket to 'x.x.x.x:53': Permission denied
Apr 20 05:33:24 Unable to reuse port, falling back to original bind
```

I thought it might be the systemd so i butchered the service file in various ways to no effect. I thought maybe some capability was missing so i changed it to `CapabilityBoundingSet=~` in order to grant all but continue to receive the same `Permission denied` error.

**Comment by rgacogne** [comment]
It looks like a duplicate of #1715, which was fixed in master by #5120. The fix has not been backported to 4.0.x yet, though.

**Comment by thechile** [comment]
OK thank you. Same thing for sure. Will close and hope this makes it into next release.
